### DISCUSSION TITLE
How can I use computer to control other computer at the same time?

### DISCUSSION THREAD

**Discussion by HenryDu8133** [cc-tweaked/CC-Tweaked#1898] [Q&A] [answered]
How can I use computer to control other computer at the same time? I try to use rednet but it can only send message but cna't send command . If you can give a detailed solution or code, that's even better. Thanks for your help!

**Comment by zyxkad** [comment, marked as answer]
You need a program on the receiver side, and make the program execute the command you have sent.

An easy example will be:

```lua
while true do
  local _, command = rednet.receive()
  local parsedCommand, err = load(command)
  if parsedCommand then
    local result = {parsedCommand()}
    -- and send back the result if you want
  end
end
```

The controller just send or broadcast Lua commands.
If you want execute shell command, just use `shell.run` instead of `load`

Note: In actual use, you may want define protocol, and add encrypt & signature to keep your computer in secure

**Reply by HenryDu8133** [reply]
Thank you very much !You help me a lot!

**Reply by HenryDu8133** [reply]
Now I've managed to control the computer remotely, but how can I achieve typing in the computer from the host computer when I need to type in certain links?

**Reply by zyxkad** [reply]
You need a fake term and forward events from one computer to another.

[This](https://github.com/zyxkad/cc/blob/master/ws/wsd.lua#L225) is my example of a fake term that forward a program's output to web browser through websocket. It's simple and not optimized, and if you search on web there should be more examples you can look at.
